src/lib/registration/widgets/registration_code.rb

```diff
@@ -67,6 +64,16 @@ def register
           return true
         end
 
+        if Registration.is_registered?
+          log.info("The system is already registered so skipped registration.")
```

Comment:
We could warn the user, just logged by now because it should not happen and also we could just check it out of the widget rendering the label of the product instead of a input field.